src/driver/src/DefaultDriver.cpp

```diff
@@ -188,58 +193,85 @@ std::string normalizeOutputFile(const std::string& str) {
+Genny run modes
+    normal
+        Run the workload normally; default mode if no run mode is specified
+
+    dry-run
+        Exit before the run step -- this may still make network connections during workload initialization
+
+    evaluate
+        Print the evaluated YAML workload file with minimal validation
+
+    list-actors
+        List all actors available for use
+    )";
+
+    progDescription.add_options()
+            ("help,h",
+             "Show help message")
+            ("run-mode", po::value<std::string>()->default_value("normal"),
+             runModeHelp)
+            ("list-actors",
+                    "DEPRECATED. Please use the \"list-actors\" run mode.")
+            ("dry-run",
+                    "DEPRECATED. Please the \"dry-run\" run mode.")
+            ("metrics-format,m",
              po::value<std::string>()->default_value("csv"),
              "Metrics format to use")
-        ("metrics-output-file,o",
-            po::value<std::string>()->default_value("/dev/stdout"),
-            "Save metrics data to this file. Use `-` or `/dev/stdout` for stdout.")
-        ("workload-file,w",
-            po::value<std::string>(),
-            "Path to workload configuration yaml file. "
-            "Paths are relative to the program's cwd. "
-            "Can also specify as first positional argument.")
-        ("mongo-uri,u",
-            po::value<std::string>()->default_value("mongodb://localhost:27017"),
-            "Mongo URI to use for the default connection-pool.")
-    ;
+            ("metrics-output-file,o",
+             po::value<std::string>()->default_value("/dev/stdout"),
+             "Save metrics data to this file. Use `-` or `/dev/stdout` for stdout.")
+            ("workload-file,w",
+             po::value<std::string>(),
+             "Path to workload configuration yaml file. "
+             "Paths are relative to the program's cwd. "
+             "Can also specify as the last positional argument.")
+            ("mongo-uri,u",
+             po::value<std::string>()->default_value("mongodb://localhost:27017"),
+             "Mongo URI to use for the default connection-pool.");
 
     positional.add("workload-file", -1);
 
     auto run = po::command_line_parser(argc, argv)
-        .options(description)
-        .positional(positional)
-        .run();
+            .options(progDescription)
+            .positional(positional)
+            .run();
     // clang-format on
 
     {
         auto stream = std::ostringstream();
-        stream << description;
+        stream << progDescription;
         this->description = stream.str();
     }
 
     po::variables_map vm;
     po::store(run, vm);
     po::notify(vm);
 
+    const auto runModeStr = vm["run-mode"].as<std::string>();
+
+    if (vm.count("list-actors") >= 1 || runModeStr == "list-actors")
+        this->runMode = RunMode::kListActors;
+
+    if (vm.count("dry-run") >= 1 || runModeStr == "dry-run")
```

Comment:
[question] Is it worth making the `--list-actors` and `--run-mode` conflict or do we not care about being able to do inane things like `./build/src/driver/genny --list-actors --run-mode dry-run src/workloads/docs/HelloWorld.yml`?

Reply:
Hm, would you be open to deleting the non-run-mode version here and in DSI? I'll of course also send out an email.

Reply:
It feels like maybe more trouble than its worth to do in this ticket right now to remove those command line options. We should get the DSI code and any other scripting switched over to using the new command line options in the next iteration though.

Reply:
My vote would be to remove the old options - seems easy to change dsi and genny at the same time and this will do a lot to make the options more sane.